src/jsdocUtils.js

```diff
@@ -126,17 +126,45 @@ const hasDefinedTypeReturnTag = (tag) => {
   return true;
 };
 
+const tagsWithOptionalType = [
+  'augments', 'extends',
+  'class', 'constructor',
+  'constant', 'const',
+  'enum',
+  'implements',
+  'member', 'var',
+  'module',
+  'namespace',
+  'param', 'arg', 'argument',
+  'property', 'prop',
+  'returns', 'return',
+  'throws', 'exception',
+  'type',
+  'typedef',
+  'yields', 'yield',
+
+  // GCC specific
+  'package',
+  'private',
+  'protected',
+  'public',
+  'static',
```

Comment:
Let's only add these if our new proposal for `settings.jsdoc.mode` is "gcc" (see #356 ).

Reply:
When it hits, definitely. You think they should be removed from the list for now?
I believe it's better to leave them be for now not to break this bit of backwards compatibility.